src/uptane/legacysecondary.cc

```diff
@@ -0,0 +1,41 @@
+#include "uptane/legacysecondary.h"
+
+#include <boost/algorithm/hex.hpp>
+#include <boost/filesystem.hpp>
+
+#include "crypto.h"
+#include "logger.h"
+
+namespace Uptane {
+LegacySecondary::LegacySecondary(const SecondaryConfig& sconfig_in) : ManagedSecondary(sconfig_in) {
+  boost::filesystem::create_directories(sconfig.firmware_path.parent_path());
+}
+
+bool LegacySecondary::storeFirmware(const std::string& target_name, const std::string& content) {
```

Comment:
Firmware might be very big and store it in memory may consume all of free memory.